tests/integration/ha_tests/application-charm/src/charm.py

```diff
@@ -94,10 +107,12 @@ def _count_writes(self) -> int:
     def _on_clear_continuous_writes_action(self, _) -> None:
         """Clears database writes."""
         self._stop_continuous_writes()
-        with psycopg2.connect(
-            self._connection_string
-        ) as connection, connection.cursor() as cursor:
-            cursor.execute("DROP TABLE continuous_writes;")
+        for attempt in Retrying(stop=stop_after_delay(60), wait=wait_fixed(3)):
+            with attempt:
+                with psycopg2.connect(
+                    self._connection_string
+                ) as connection, connection.cursor() as cursor:
+                    cursor.execute("DROP TABLE continuous_writes;")
```

Comment:
Sometimes tests failed due to this happening in a read only transaction. Retrying to make certain the table will be dropped.